### ISSUE TITLE
ruby version important?

### ISSUE THREAD

**Issue by stedy** [rossf7/elasticrawl#3]
Just spun up a new hadoop cluster on Amazon EMR with default settings and got this:

```
[hadoop@ip-172-31-37-54 elasticrawl-1.1.0-linux-x86_64]$ ./elasticrawl parse CC-MAIN-2014-49 --max-segments 2 --max-files 3
/home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/gems/aws-sdk-v1-1.60.2/lib/aws/core/client.rb:375:in `return_or_raise': AWS::Errors::Base (AWS::Errors::Base)
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/gems/aws-sdk-v1-1.60.2/lib/aws/core/client.rb:476:in `client_request'
        from (eval):3:in `head_object'
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/gems/aws-sdk-v1-1.60.2/lib/aws/s3/s3_object.rb:296:in `head'
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/gems/aws-sdk-v1-1.60.2/lib/aws/s3/s3_object.rb:273:in `exists?'
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/gems/elasticrawl-1.1.0/lib/elasticrawl/crawl.rb:106:in `uncompress_file'
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/gems/elasticrawl-1.1.0/lib/elasticrawl/crawl.rb:99:in `warc_paths'
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/gems/elasticrawl-1.1.0/lib/elasticrawl/crawl.rb:51:in `create_segments'
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/gems/elasticrawl-1.1.0/lib/elasticrawl/crawl.rb:42:in `has_segments?'
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/gems/elasticrawl-1.1.0/bin/elasticrawl:48:in `parse'
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/gems/elasticrawl-1.1.0/bin/elasticrawl:134:in `<top (required)>'
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/bin/elasticrawl:23:in `load'
        from /home/hadoop/elasticrawl-1.1.0-linux-x86_64/lib/vendor/ruby/2.1.0/bin/elasticrawl:23:in `<main>'
```

It seems like its a ruby error but I updated ruby to 2.2 and got the same issue. Am I way off?

**Comment by rossf7** [comment]
Hi, I don't think the problem is the Ruby version. The deploy package includes a Ruby 2.1 interpreter and it uses that. It looks like the problem is downloading the WARC paths file in the Common Crawl bucket. This file is used to work out which segments are in the crawl.  

I think it could be an AWS permissions problem. Did the init command work OK? Also do you have the s3cmd tool?  If so does this command work using your access keys?

s3cmd get s3://aws-publicdatasets/common-crawl/crawl-data/CC-MAIN-2014-49/warc.paths.gz

**Comment by stedy** [comment]
Installed s3cmd and the s3cmd get command mentioned works fine. The mentioned init command completes fine also. I have started multiple clusters and destroyed and remade all my keys and tried with different IAMs with all iterations giving the same error.

**Comment by rossf7** [comment]
Thanks for trying all these steps and I'm sorry they didn't work. I've released a new version (1.1.2) that has better error handling for the S3 API calls.  Could you install it and try again please?

Also could you tell me which AMI you're using? I can then run some tests using the same VM.

**Comment by stedy** [comment]
That works, now for some experiments!
